### PR TITLE
Added kDoc for Chat interface.

### PR DESCRIPTION
**Pull request by marcin-cebo**

### CODE REVIEW

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 1]
```diff
@@ -25,11 +25,55 @@ import com.pubnub.kmp.CustomObject
...
-    val pubNub: PubNub // todo change to `sdk` like in TS
+
+    /**
+     * Allows you to access any Kotlin SDK method. For example, if you want to call a method available in the
+     * App Context API, you'd use chat.pubNub.getAllUUIDMetadata().
+     */
+    val pubNub: PubNub
+
+    /**
+     * Unique User ID that becomes your app's current user. It's a string of up to 92 characters that identifies
```
[the remark this answers is not in the document]
Comment (wkal-pubnub) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:51]: it's also not a string

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 2]
```diff
@@ -136,58 +339,166 @@ interface Chat {
...
         limit: Int? = null,
         page: PNPage? = null,
         filter: String? = null,
         sort: Collection<PNSortKey<PNMembershipKey>> = listOf(),
     ): PNFuture<MarkAllMessageAsReadResponse>
 
+    /**
+     * Retrieves all channels referenced in the [Channel.sendText] that match the provided 3-letter string from
+     * your app's keyset.
+     * @param text At least a 3-letter string typed in after # with the channel name you want to reference.
```
Comment (wkal-pubnub) [inline pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:436]: the implementation suggests that `text` should contain the "#"...
in any case, I think we should hide this function until we have MessageDraft as the implementation will change and we don't want to support it indefinitely

same with user suggestions

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 3]
```diff
@@ -136,58 +339,166 @@ interface Chat {
...
+     * e.g. EventContent.Receipt::class, EventContent.TextMessageContent::class
+     * @param channelId Channel to listen for new events.
+     * @param customMethod An optional custom method for emitting events. If not provided, defaults to null.
+     * @param callback A lambda function that is called with an Event<T> as its parameter.
+     * It defines the custom behavior to be executed whenever an event is detected on the specified channel.
+     *
+     * @return AutoCloseable Interface you can call to stop listening for new messages and clean up resources when they
+     * are no longer needed by invoking the close() method.
+     */
     fun <T : EventContent> listenForEvents(
```
Comment (wkal-pubnub) [inline pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:355]: the extension function with the same name is not documented

File: pubnub-chat-impl/src/commonMain/kotlin/com/pubnub/chat/internal/ChatInternal.kt [hunk 4]
```diff
@@ -23,4 +27,17 @@ interface ChatInternal : Chat {
     ): PNFuture<Pair<PNRemoveMessageActionResult, Channel>>
 
     fun restoreThreadChannel(message: Message): PNFuture<PNMessageAction?>
+
+    fun createChannel(
```
Comment (wkal-pubnub) [inline pubnub-chat-impl/src/commonMain/kotlin/com/pubnub/chat/internal/ChatInternal.kt:31]: nice catch that these are internal! I didn't see it

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:51]: Good point! I will fix that.

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 5]
```diff
@@ -41,18 +85,55 @@ interface Chat {
...
+     *
+     * @return [PNFuture] containing [User] object.
+     */
     fun getUser(userId: String): PNFuture<User?>
 
+    /**
+     * Returns a paginated list of all users and their details
+     *
+     * @param filter Expression used to filter the results. Returns only these users whose properties satisfy the
+     * given expression are returned. The filtering language is defined in docs.
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:101]: I will add.

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 6]
```diff
@@ -62,52 +143,138 @@ interface Chat {
         type: String? = null,
     ): PNFuture<User>
 
+    /**
+     * Deletes a [User] with or without deleting its historical data from the App Context storage.
+     *
+     * @param id Unique user identifier.
+     * @param soft Define if you want to permanently remove user metadata. The user metadata gets permanently deleted
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:150]: ok, I will change.

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 7]
```diff
@@ -62,52 +143,138 @@ interface Chat {
...
+     *
+     * @return [PNFuture] containing Collection of [User.id].
+     */
     fun whoIsPresent(channelId: String): PNFuture<Collection<String>>
 
+    /**
+     * Constructs and sends events with your custom payload.
+     *
+     * @param channelId Channel where you want to send the events.
+     * @param payload Type of events. Use custom for full control over event payload and emitting method.
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:252]: I will change it.

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 8]
```diff
@@ -136,58 +339,166 @@ interface Chat {
         membershipCustom: CustomObject? = null,
     ): PNFuture<CreateGroupConversationResult>
 
+    /**
+     * Lets you watch a selected channel for any new custom events emitted by your chat app.
+     *
+     * @param type Reified type parameter bounded by the EventContent interface, allowing access to type information at runtime
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:345]: I will fix that.

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt [hunk 9]
```diff
@@ -136,58 +339,166 @@ interface Chat {
...
+     * e.g. EventContent.Receipt::class, EventContent.TextMessageContent::class
+     * @param channelId Channel to listen for new events.
+     * @param customMethod An optional custom method for emitting events. If not provided, defaults to null.
+     * @param callback A lambda function that is called with an Event<T> as its parameter.
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:349]: done

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt (hunk 2 above)
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:436]: Good point. I will move them to ChatInternal.

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt (hunk 3 above)
Comment (marcin-cebo) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt:355]: Done

**Pull request merged**